common/src/test/java/io/netty/util/concurrent/DefaultThreadFactoryTest.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.util.concurrent;
+
+import org.junit.Test;
+
+import java.security.Permission;
+import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+public class DefaultThreadFactoryTest {
+    @Test(timeout = 2000)
+    public void testDescendantThreadGroups() throws InterruptedException {
+        final SecurityManager current = System.getSecurityManager();
+
+        try {
+            // install security manager that only allows parent thread groups to mess with descendant thread groups
+            System.setSecurityManager(new SecurityManager() {
+                @Override
+                public void checkAccess(ThreadGroup g) {
+                    final ThreadGroup source = Thread.currentThread().getThreadGroup();
+
+                    if (source != null) {
+                        if (!source.parentOf(g)) {
```

Comment:
@normanmaurer I don't think that gives the right semantics. The value of `source` can be `null` (yes, this can happen if the thread is stopped). In that case, there is no need to check permissions, we can just return. So rewriting this to 

```java
if (source != null && !source.parentOf(g)) {
    throw new SecurityException("source group is not an ancestor of the target group");
}
super.checkAccess(g);
```

would lead to check permissions when `source` is `null`. I think it's okay as is?